### ISSUE TITLE
Import from file not working Android 14

### ISSUE THREAD

**Issue by mxg318** [osmandapp/OsmAnd#20778]
### Description

Unable to import from file on Android 14. Tablet and phone. Works ok on Android 9 & 10 older tablet and phone.

### Steps to reproduce

After doing an "export from file" using the settings/export to file/settings&my places and the selecting "import from file" nothing happens, it just returns to the settings menu

### Actual result

Nothing happens just returns to the settings menu

### Expected result

It should open a selection box showing settings my places resources

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version:
Android/iOS version: 14
Device model: Oppo x3 pro & Samsung A8 tablet
Crash-Logs: ?  No crash to log
```

**Comment by yuriiurshuliak** [comment]
Could you please attach the file you're trying to import that is causing the issue?

**Comment by mejo-** [comment]
Probably the same issue as reported in #20718

**Comment by mxg318** [comment]
I cannot attach the file here, when I try to it says file type not supported, it is just the usual export file generated by Osmand, Export_2024-09-10.osf

**Comment by mxg318** [comment]
[Export_2024-09-10.zip](https://github.com/user-attachments/files/16937173/Export_2024-09-10.zip)

I read #20718, it suggests zipping the file which I have done and it is now attached here.

**Comment by mxg318** [comment]
Were you importing to a phone that was using Android 14, that is my issue. It imports correctly on all earlier Android versions.

I have no idea what the error message you mention means or how to rectify it, can you elaborate please.

**Comment by Gurthurb** [comment]
I'm experiencing the same problem, on Android 14 (trying to load a file exported from an Android 13 device). Nothing happens, OsmAnd doesn't even appear to attempt importing?

**Comment by mxg318** [comment]
For me it is just a problem on Android 14, I can restore the export on earlier versions of Android. I am able to save each track file as a gpx and open them in 14 but only one at a time.






---- On Mon, 16 Sep 2024 07:25:58 +1000 ***@***.*** wrote ----


DuckDuckGo removed one tracker. More
 
Deactivate

I'm experiencing the same problem, on Android 14 (trying to load a file exported from an Android 13 device). Nothing happens, OsmAnd doesn't even appear to attempt importing?

**Comment by Gurthurb** [comment]
Answered via email, but can't see it here (yet?), so I'll post the reply as a comment here as well:

It's the same for me, I can import a file containing recorded tracks in Android 13, but when importing in Android 14, nothing will happen at all if the file contains recorded tracks.

I exported a new file containing ONLY recorded tracks (from Android 13), and as suspected that file can't be imported in Android 14 neither, nothing happens.

Hence it isn't a matter of combinations of data categories in the file, it's specifically recorded tracks, and apparently other data categories as well.

The most relevant lines in OsmAnd's logcat dump, it appears to be a pathing issue withing the zip:

```shell
09-16 12:38:55.230  7149  7162 E net.osmand: SettingsHelper Failed to read next entry {AsyncTask #1}
09-16 12:38:55.230  7149  7162 E net.osmand: java.util.zip.ZipException: Invalid zip entry path: /tracks/rec/2020-08-15_20-00_Sat.gpx
09-16 12:38:55.230  7149  7162 E net.osmand:     at com.android.internal.os.SafeZipPathValidatorCallback.onZipEntryAccess(SafeZipPathValidatorCallback.java:52)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1695)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile$Source.<init>(ZipFile.java:1436)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile$Source.get(ZipFile.java:1389)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:769)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile.<init>(ZipFile.java:283)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile.<init>(ZipFile.java:239)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile.<init>(ZipFile.java:187)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.zip.ZipFile.<init>(ZipFile.java:156)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.SettingsImporter.updateFilesInfo(SettingsImporter.java:86)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.SettingsImporter.getItemsFromJson(SettingsImporter.java:66)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.SettingsImporter.processItems(SettingsImporter.java:106)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.SettingsImporter.collectItems(SettingsImporter.java:37)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.ImportFileTask.doInBackground(ImportFileTask.java:103)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.ImportFileTask.doInBackground(ImportFileTask.java:24)
09-16 12:38:55.230  7149  7162 E net.osmand:     at android.os.AsyncTask$3.call(AsyncTask.java:394)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.concurrent.FutureTask.run(FutureTask.java:264)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.lang.Thread.run(Thread.java:1012)
09-16 12:38:55.230  7149  7162 E net.osmand: SettingsHelper Failed to collect items from: Export_2024-09-16.osf {AsyncTask #1}
09-16 12:38:55.230  7149  7162 E net.osmand: java.lang.IllegalArgumentException: No items
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.SettingsImporter.processItems(SettingsImporter.java:109)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.SettingsImporter.collectItems(SettingsImporter.java:37)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.ImportFileTask.doInBackground(ImportFileTask.java:103)
09-16 12:38:55.230  7149  7162 E net.osmand:     at net.osmand.plus.settings.backend.backup.ImportFileTask.doInBackground(ImportFileTask.java:24)
09-16 12:38:55.230  7149  7162 E net.osmand:     at android.os.AsyncTask$3.call(AsyncTask.java:394)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.concurrent.FutureTask.run(FutureTask.java:264)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
09-16 12:38:55.230  7149  7162 E net.osmand:     at java.lang.Thread.run(Thread.java:1012)
```

**Comment by Michael21011** [comment]
I have the same issue on android 12.
I cannot import Gpx file with points, osmand just do nothing after select file. Issue was started in beta 4.9. version 4.8.6 is ok

[postboxes(11).gpx.zip](https://github.com/user-attachments/files/17080718/postboxes.11.gpx.zip)

**Comment by vshcherb** [comment]
It's an old issue with broken Kotlin Zip reader which is fixed in 4.9.2 beta, 4.9 public release is planning soon
